spec/models/like_spec.rb

```diff
@@ -0,0 +1,16 @@
+require 'rails_helper'
+
+RSpec.describe Like, type: :model do
+  it 'is valid with a post_id' do
+    user = User.create(name: 'John', posts_counter: 0)
+    post = Post.create(title: 'Test Post', comments_counter: 0, likes_counter: 0, author: user)
+    like = Like.new(post:, author: user)
+    expect(like).to be_valid
+  end
+
+  it 'is invalid without a post_id' do
+    user = User.create(name: 'John', posts_counter: 0)
+    like = Like.new(author: user)
+    expect(like).to_not be_valid
+  end
```

Comment:
- [ ] kindly add a test to prove the private method `increment_like_counter` works, you can do that by calling the `after_save method`